client/go/commands/node_add.go

```diff
@@ -0,0 +1,163 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package commands
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/heketi/heketi/apps/glusterfs"
+	"github.com/heketi/heketi/utils"
+	"github.com/lpabon/godbc"
+	"net/http"
+	"os"
+	"time"
+)
+
+type NodeAddCommand struct {
+	Cmd
+	options            *Options
+	zone               int
+	managmentHostNames string
+	storageHostNames   string
+	clusterId          string
+}
+
+func NewNodeAddCommand(options *Options) *NodeAddCommand {
+
+	godbc.Require(options != nil)
+
+	cmd := &NodeAddCommand{}
+	cmd.name = "add"
+	cmd.options = options
+	cmd.flags = flag.NewFlagSet(cmd.name, flag.ExitOnError)
+	cmd.flags.IntVar(&cmd.zone, "zone", 0, "The zone in which the node should reside")
+	cmd.flags.StringVar(&cmd.clusterId, "cluster", "", "The cluster in which the node should reside")
+	cmd.flags.StringVar(&cmd.managmentHostNames, "managment-host-name", "", "Managment host name")
+	cmd.flags.StringVar(&cmd.storageHostNames, "storage-host-name", "", "Storage host name")
+
+	//usage on -help
+	cmd.flags.Usage = func() {
+		fmt.Println(`
+Node add is a command used for adding a node to the specified cluster.
+
+USAGE
+	heketi node add [options]
+
+OPTIONS`)
+
+		//print flags
+		cmd.flags.PrintDefaults()
+		fmt.Println(`
+EXAMPLES
+	Add a node to heketi
+		$ heketi -server http://localhost:8080 node add \
+	  		 -zone 3 \
+	  		 -cluster 3e098cb4407d7109806bb196d9e8f095 \
+	  		 -managment-host-name node1-manage.gluster.lab.com \
+	  		 -storage-host-name node1-storage.gluster.lab.com`)
+	}
+	godbc.Ensure(cmd.flags != nil)
+	godbc.Ensure(cmd.name == "add")
+
+	return cmd
+}
+
+func (a *NodeAddCommand) Name() string {
+	return a.name
+
+}
+
+func (a *NodeAddCommand) Exec(args []string) error {
+
+	//parse args
+	a.flags.Parse(args)
+
+	//ensure we have Url
+	if a.options.Url == "" {
+		fmt.Fprintf(stdout, "You need a server!\n")
+		os.Exit(1)
```

Comment:
We probably should not have any of the command functions call os.Exit().  Instead it should return an error